File: plugin/testDataOutput/shared/lambda.swift
//Package: com.test
//Converted using Kwift2

import Foundation



public func lambdaTest() -> Void {
    takesLambda(lambda: { (integer, someValue) in 
        print("Hello!")
    })
    takesLambda{ (integer, someValue) in 
        print("Hello!")
    }
    takesLambda({ [unowned self] (integer, someValue) in 
        print("Hello!")
    })
}
 
 

public func takesLambda(lambda: @escaping (_ integer: Int32, _ someValue: String) -> Void) -> Void {
}
 
 

public func takesPainfulLambda(lambda: @escaping (_ integer: Int32, _ someValue: @escaping (String) -> Void) -> Void) -> Void {
}
 
 

public func takesEscapingLambda(lambda: @escaping (_ integer: Int32, _ someValue: String) -> Void) -> Void {
}
 
 

public func takesEscapingLambda2(lambda: (@escaping (_ integer: Int32, _ someValue: String) -> Void)) -> Void {
}
 
 

public class ConstructorHasEscaping {
    
    public var lambda:  (_ integer: Int32, _ someValue: String) -> Void
    
    
    public init(lambda: @escaping (_ integer: Int32, _ someValue: String) -> Void) {
        self.lambda = lambda
    }
    convenience public init(_ lambda: @escaping (_ integer: Int32, _ someValue: String) -> Void) {
        self.init(lambda: lambda)
    }
}
 
 
 

public class ConstructorHasEscaping2 {
    
    public var lambda: (@escaping (_ integer: Int32, _ someValue: String) -> Void)
    
    
    public init(lambda: (@escaping (_ integer: Int32, _ someValue: String) -> Void)) {
        self.lambda = lambda
    }
    convenience public init(_ lambda: (@escaping (_ integer: Int32, _ someValue: String) -> Void)) {
        self.init(lambda: lambda)
    }
}
 
 

public class HeavenHelpUsAll {
    
    public var lambda:  (_ integer: Int32, _ someValue: @escaping (Int32) -> Void) -> Void
    
    
    public init(lambda: @escaping (_ integer: Int32, _ someValue: @escaping (Int32) -> Void) -> Void) {
        self.lambda = lambda
    }
    convenience public init(_ lambda: @escaping (_ integer: Int32, _ someValue: @escaping (Int32) -> Void) -> Void) {
        self.init(lambda: lambda)
    }
}
 

File: plugin/testData/shared/lambda.kt
package com.test

fun lambdaTest() {
    takesLambda(lambda = { integer, someValue ->
        println("Hello!")
    })
    takesLambda { integer, someValue ->
        println("Hello!")
    }
    takesLambda(label@ @unownedSelf { integer, someValue ->
        println("Hello!")
    })
}

fun takesLambda(lambda: @escaping() (integer: Int, someValue: String) -> Unit) {

}

fun takesPainfulLambda(lambda: @escaping() (integer: Int, someValue: @escaping() (String) -> Unit) -> Unit) {

}

fun takesEscapingLambda(lambda: @escaping() (integer: Int, someValue: String) -> Unit) {

}

fun takesEscapingLambda2(lambda: (@escaping() (integer: Int, someValue: String) -> Unit)) {

}

class ConstructorHasEscaping(val lambda: @escaping (integer: Int, someValue: String) -> Unit) {
}


class ConstructorHasEscaping2(val lambda: (@escaping (integer: Int, someValue: String) -> Unit)) {
}

class HeavenHelpUsAll(val lambda: @escaping() (integer: Int, someValue: @escaping() (Int) -> Unit) -> Unit) {
}
